### ISSUE TITLE
Tests do not run

### ISSUE THREAD

**Issue by etothepii** [kondratyev-nv/vscode-python-test-adapter#232]
I have a couple of tests set up using pytest.

![image](https://user-images.githubusercontent.com/2609402/111461107-8f0f7400-86fb-11eb-8082-db9dee791f6e.png)

When I click play they go blue with the elipsis

![image](https://user-images.githubusercontent.com/2609402/111461188-a9e1e880-86fb-11eb-97aa-2c0f9ec64820.png)

But then return.

This appears to be happening on both my and my co-founders computers across multiple projects and appears to have started occurring sometime during last night (everything was working at 2000 GMT on Tuesday 16th March. 

I can't see that a new version has been released, but as it is happening on multiple computers across multiple projects I assume something external has changed.

setting.json looks like:

    {
        "python.pythonPath": "SOME_CONDA_ENVIROMENT",
        "python.testing.pytestArgs": [
            "remove-white-background"
        ],
        "python.testing.unittestEnabled": false,
        "python.testing.nosetestsEnabled": false,
        "python.testing.pytestEnabled": true
    }

The python test adapter log appears free of errors:

    2021-03-17T11:34:37.107Z INFO pytest at 'project-name': Reading configuration for workspace project-name
    2021-03-17T11:34:37.108Z  DBG pytest at 'project-name': usingNewInterpreterStorage feature flag is 'true'
    2021-03-17T11:34:37.108Z INFO pytest at 'project-name': Using auto-detected pythonPath SOME_CONDA_ENVIROMENT
    2021-03-17T11:34:37.109Z INFO pytest at 'project-name': Running tests using python path 'SOME_CONDA_ENVIROMENT' in PROJECT_FOLDER
    2021-03-17T11:34:37.109Z INFO pytest at 'project-name': Environment variables file PROJECT_FOLDER/.env does not exist
    2021-03-17T11:34:37.113Z INFO pytest at 'project-name': Running pytest with arguments: --rootdir=PROJECT_FOLDER, --junitxml=/var/folders/kk/m58vb4996x10jnrg4n7kcchc0000gn/T/tmp-1272-buZvkt6qIWZI, --override-ini, junit_logging=all, --override-ini, junit_family=xunit1, folder-name
    2021-03-17T11:34:37.113Z INFO pytest at 'project-name': Running pytest as a Python module
    2021-03-17T11:34:42.571Z INFO pytest at 'project-name': Test execution completed

The tests run correctly with `pytest .`

*Update* 

Errors can be caused to appear on uncompilable code.

![image](https://user-images.githubusercontent.com/2609402/111463110-fe866300-86fd-11eb-8f2f-74a3e9614e66.png)

**Comment by voidz0r** [comment]
I'm having the same error since the last VSCode update,  looks like an integration issue. My tests are executing correctly with pytest cli

**Comment by Keyeoh** [comment]
Another one here with the same error.

 * Simple project with pipenv.

![image](https://user-images.githubusercontent.com/1183823/111497014-0d423980-8741-11eb-819b-1b827d5787f9.png)

 * Tests run correctly using pytest from terminal.

![image](https://user-images.githubusercontent.com/1183823/111497423-627e4b00-8741-11eb-96d6-1fcc51849404.png)

 * Extension seems not to do anything.

![image](https://user-images.githubusercontent.com/1183823/111497200-35319d00-8741-11eb-86d5-36d5b1ce0ca5.png)

 * Output messages are not very informative.

![image](https://user-images.githubusercontent.com/1183823/111497274-424e8c00-8741-11eb-86a0-cd2e19067198.png)

**Comment by BrendanSimon** [comment]
Yep.  Seeing the same thing.  I downgraded to vscode 1.53 and things are working again.
Same thing happens with Insdiers build too.

I did notice that the standard vscode testing tree works ok.

![image](https://user-images.githubusercontent.com/472245/111547531-69eb3600-87cd-11eb-9672-77e5d39be3fb.png)

**Comment by ws1088** [comment]
I spent the last 3 hours trying to figure out the same issue. Looks like if the test is the same directory as the workspace then the test will run, all other tests will turn "blue" and then "grey", same as all reporters above. I am using 1.55 and insider with this issue. I remember last time I used 1.53 and it is working well. FYI

**Comment by SophiaZ12** [comment]
I have the same issue. I noticed that it works with Unittest and the problem only occurs with Pytest.

**Comment by etothepii** [comment]
I have confirmed that regressing to v1.53.2 solves the problem, unfortunately VSCode is a little "upgrade happy" so will likely upgrade me and break again.

**Comment by kondratyev-nv** [comment]
Sorry for the wait 😥 I can confirm the issue. Will try to fix it today.

**Comment by etothepii** [comment]
No need to apologise! A short term work around has been identified.

**Comment by Necromancerx** [comment]
Thank you for working hard @kondratyev-nv <3

**Comment by kondratyev-nv** [comment]
I've published a new version 0.6.7. Please, update and let me know if the issue persists.

**Comment by BrendanSimon** [comment]
Works for me now on Win10 Pro, for both Insiders and Stable (V1.54.3).

```
Version: 1.54.3 (user setup)
Commit: 2b9aebd5354a3629c3aba0a5f5df49f43d6689f8
Date: 2021-03-15T10:55:45.459Z
Electron: 11.3.0
Chrome: 87.0.4280.141
Node.js: 12.18.3
V8: 8.7.220.31-electron.0
OS: Windows_NT x64 10.0.19042
```

**Comment by ws1088** [comment]
thanks, works for me as well. thanks again!

**Comment by Keyeoh** [comment]
![image](https://user-images.githubusercontent.com/1183823/111746311-273d6280-888e-11eb-82f3-b9cb0ebec63b.png)

It's great to see all this green again! :)

Thanks a lot @kondratyev-nv for your great work!

**Comment by farfalk** [comment]
I'm still having this issue in vscode 1.56.2 running in WSLv1(Ubuntu 20.04.2 LTS), with extension version 0.6.8, using pytest 5.4.3. All tests turn blue and seemingly run, but they all turn grey at the end.

Tests work correctly in python extension test module and from terminal, as in above comments.

Sorry to reopen this :(

**Comment by kondratyev-nv** [comment]
@farfalk I think this is might be another issue. Can you please post it as a new issue and add some more details, like testing framework (maybe plugins), some output from CLI run, and whether there are any weird messages in Python Test Adapter logs.

**Comment by farfalk** [comment]
> @farfalk I think this is might be another issue. Can you please post it as a new issue and add some more details, like testing framework (maybe plugins), some output from CLI run, and whether there are any weird messages in Python Test Adapter logs.

Sorry for the **extremely late** response, I missed the notification.
I tested it again today after more than a month, and reproduced the issue with vscode 1.58.2.
_**BUT I SOLVED IT**_, doing what follows:
1. deleted the `.pytest_cache` folder in root folder
2. deleted the `__pycache__` folder in the test folder
3. closed the vscode editor without doing anything else
4. reopened the vscode editor
5. ran the tests
6. everything turned green :)

Since this, everything works without any issue.
Reproduced the fix in all my repos, with multiple python versions.
